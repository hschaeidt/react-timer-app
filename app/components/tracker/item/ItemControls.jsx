import React, { Component, PropTypes } from 'react';

class ItemControls extends Component {
  constructor(props) {
    super(props);

    this.handleNewRange = this.handleNewRange.bind(this);
  }


  handleNewRange(e) {
    e.preventDefault();

    const from = this.from.value;
    const to = this.to.value;

    const timeFormat = /[0-9]{2}:[0-9]{2}/;

    if (from.match(timeFormat) && to.match(timeFormat)) {
      this.from.value = '';
      this.to.value = '';

      this.props.onNewRange(from, to);
    }
  }

  render() {
    const { onComplete } = this.props;

    return (
      <div className="button-group">
        <form id="handle-new-range-form" onSubmit={this.handleNewRange}>
          <input type="time" placeholder="--:--" ref={(ref) => { this.from = ref; }} />
          <input type="time" placeholder="--:--" ref={(ref) => { this.to = ref; }} />
          <button className="button primary">Add range</button>
        </form>
        <button
          id="lock-button"
          className="button alert hollow"
          onClick={onComplete}
        >
          Lock
        </button>
      </div>
    );
  }
}

ItemControls.propTypes = {
  onNewRange: PropTypes.func.isRequired,
  onComplete: PropTypes.func.isRequired,
};

export default ItemControls;
