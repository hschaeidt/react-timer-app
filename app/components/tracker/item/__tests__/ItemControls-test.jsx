import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ItemControls from '../ItemControls';

function setup() {
  const props = {
    onNewRange: jest.fn(),
    onComplete: jest.fn(),
  };

  const wrapper = mount(<ItemControls {...props} />);

  return {
    wrapper,
    props,
  };
}

describe('<ItemControls />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ItemControls onComplete={jest.fn()} onNewRange={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });

  it('should call onNewRange when submitted with valid times', () => {
    const { wrapper, props } = setup();
    wrapper.instance().from.value = '10:00';
    wrapper.instance().to.value = '12:00';
    wrapper.find('#handle-new-range-form').simulate('submit');
    expect(props.onNewRange).toHaveBeenCalledWith('10:00', '12:00');
  });
});
