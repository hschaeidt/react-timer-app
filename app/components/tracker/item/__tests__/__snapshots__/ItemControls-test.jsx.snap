exports[`<ItemControls /> renders correctly 1`] = `
<div
  className="button-group">
  <form
    id="handle-new-range-form"
    onSubmit={[Function]}>
    <input
      placeholder="--:--"
      type="time" />
    <input
      placeholder="--:--"
      type="time" />
    <button
      className="button primary">
      Add range
    </button>
  </form>
  <button
    className="button alert hollow"
    id="lock-button"
    onClick={[Function]}>
    Lock
  </button>
</div>
`;
